python/Ganga/GPIDev/Adapters/IGangaFile.py

```diff
@@ -71,6 +109,36 @@ def put(self):
         """
         raise NotImplementedError
 
+    def copyTo(self, targetPath):
+        """
+        Copy a the file to the local storage using the appropriate file-transfer mechanism
+        This will raise an excpetion if targetPath isn't set to something sensible.
+        Args:
+            targetPath (str): Target path where the file is to copied to
+        """
+        if isinstance(targetPath, str) and targetPath:
+            if regex.search(self.namePattern) is None\
+                and os.path.isfile(os.path.join(self.localDir, self.namePattern)):
+
+                if not os.path.isfile(os.path.join(targetPath, self.namePattern)):
+                    shutil.copy(os.path.join(self.localDir, self.namePattern), os.path.join(targetPath, self.namePattern))
+                else:
+                    logger.debug("Already found file: %s" % os.path.join(targetPath, self.namePattern))
+
+            # Again, cannot perform a remote glob here so have to ignore wildcards
+            else:
+                self.internalCopyTo(targetPath)
+        else:
+            raise GangaException("Cannot perform a copyTo with no given targetPath!")
```

Comment:
Did you miss this comment? It seems to have been 'outdated' by something so is not on the file review screen.